lib/timers.js

```diff
@@ -424,80 +426,142 @@ exports.enroll = function(item, msecs) {
  * DOM-style timers
  */
 
+class Timeout {
+  constructor(callback, after, args, isRepeat) {
+    after *= 1; // coalesce to number or NaN
+    if (!(after >= 1 && after <= TIMEOUT_MAX)) {
+      if (after > TIMEOUT_MAX) {
+        process.emitWarning(`${after} does not fit into` +
+                            ' a 32-bit signed integer.' +
+                            '\nTimeout duration was set to 1.',
+                            'TimeoutOverflowWarning');
+      }
+      after = 1; // schedule on next tick, follows browser behavior
+    }
+
+    this._called = false;
+    this._idleTimeout = after;
+    this._idlePrev = this;
+    this._idleNext = this;
+    this._idleStart = null;
+    this._onTimeout = callback;
+    this._timerArgs = args;
+    this._repeat = isRepeat ? after : null;
+    this._destroyed = false;
+    if (process.domain)
+      this.domain = process.domain;
+
+    this[async_id_symbol] = ++async_id_fields[kAsyncIdCounter];
+    this[trigger_async_id_symbol] = initTriggerId();
+    if (async_hook_fields[kInit] > 0)
```

Comment:
Please use braces for multi-line bodies.

Reply:
This is just relocated code. Happy to adjust though.